src/pages/workspace/companyCards/WorkspaceCompanyCardsPage.tsx

```diff
@@ -108,47 +100,68 @@ type WorkspaceCompanyCardPageProps = StackScreenProps<FullScreenNavigatorParamLi
+        Policy.openPolicyCompanyCardsPage(policyID, workspaceAccountID);
+    }, [policyID, workspaceAccountID]);
+
+    useFocusEffect(fetchCompanyCards);
+
+    const companyCards = cardFeeds?.companyCards ?? {};
+    const selectedCompanyCard = companyCards[selectedFeed] ?? {};
+    const isNoFeed = lodashIsEmpty(selectedCompanyCard);
+    const isPending = selectedCompanyCard?.pending;
+    const isFeedAdded = !isPending && !lodashIsEmpty(selectedCompanyCard);
+    const isLoading = !cardFeeds || cardFeeds.isLoading;
 
     // TODO: use data form onyx instead of mocked one when API is implemented
     // const [cardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${workspaceAccountID}_${selectedFeed}`);
     const cardsList = mockedCards ?? {};
 
-    // TODO correct Onyx flag should be defined in separate PR for "Pending State with No Other Feeds"
-    const isFeedAdded = true;
-
     return (
         <AccessOrNotFoundWrapper
             policyID={route.params.policyID}
             featureName={CONST.POLICY.MORE_FEATURES.ARE_COMPANY_CARDS_ENABLED}
         >
-            <WorkspacePageWithSections
-                shouldUseScrollView={!isFeedAdded}
-                icon={Illustrations.CompanyCard}
-                headerText={translate('workspace.common.companyCards')}
-                route={route}
-                guidesCallTaskID={CONST.GUIDES_CALL_TASK_IDS.WORKSPACE_COMPANY_CARDS}
-                shouldShowOfflineIndicatorInWideScreen
-                includeSafeAreaPaddingBottom
-                showLoadingAsFirstRender={false}
-            >
-                {isFeedAdded && (
-                    <WorkspaceCompanyCardsListHeaderButtons
-                        policyID={policyID}
-                        selectedFeed={selectedFeed}
-                    />
-                )}
-                {!isFeedAdded && <WorkspaceCompanyCardPageEmptyState route={route} />}
-                {isFeedAdded && <WorkspaceCompanyCardsList cardsList={cardsList} />}
-            </WorkspacePageWithSections>
+            {isLoading && (
+                <ActivityIndicator
+                    size={CONST.ACTIVITY_INDICATOR_SIZE.LARGE}
+                    style={styles.flex1}
+                    color={theme.spinner}
+                />
+            )}
+            {!isLoading && (
+                <WorkspacePageWithSections
+                    shouldUseScrollView={!isFeedAdded}
+                    icon={Illustrations.CompanyCard}
+                    headerText={translate('workspace.common.companyCards')}
+                    route={route}
+                    guidesCallTaskID={CONST.GUIDES_CALL_TASK_IDS.WORKSPACE_COMPANY_CARDS}
+                    shouldShowOfflineIndicatorInWideScreen
+                    includeSafeAreaPaddingBottom
+                    showLoadingAsFirstRender={false}
+                >
+                    {isFeedAdded && (
+                        <WorkspaceCompanyCardsListHeaderButtons
```

Comment:
```suggestion
                    {(isFeedAdded || isPending) && (
                        <WorkspaceCompanyCardsListHeaderButtons
```
We need to show the header items for pending state too.

Reply:
Oh yeah, thanks for catching this, forgot to change it after changing isFeedAdded logic